### PR TITLE
Add new fitting function for Teixeira model

### LINKED ISSUE #36502: Teixeira Water for IqtFit tab
**Description**
We want to implement a slightly more complex version of the Teixeira Water function in Mantid. We can call it `TeixeiraWaterIqt` for now. We then want to make this function available in the IqtFit tab of the Data Analysis interface.

The equation is equation 1 in the attached paper, but without the exponential term. See also Equation 2 and 3 for the full equation.

[TeixeiraWater.pdf](https://github.com/mantidproject/mantid/files/13527948/TeixeiraWater.pdf)

From Spencer: 
```
Eqtn 2 is the equivalent of 2 peaks with FWHM & Intensities linked

I also attach a Py function of it that I put together but it didn't work too well. I may have done something wrong. 
It does need to get Q from the ws being fitted
```

[TeixeiraWaterIqt.zip](https://github.com/mantidproject/mantid/files/13527954/TeixeiraWaterIqt.zip)

**Acceptance Criteria**
- [x] This `TeixeiraWaterIqt` function is implemented inside Mantid **Not blocked - go ahead!**
- [ ] This TW function is accessible from the IqtFit tab under the "Fit type" option

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
This PR adds a fitting function for the intermediary scattering function of quasielastic spectra by using the Teixeira formalism for water. Relevant links can be found in #36502 
#### Summary of work

A new fitting function has been added, along with test and documentation. 

Part of #36502 . 

### To test:
- Build the docs, you should find new fitting function `TeixeiraWaterIqt` on `Fit Functions` list section of documentation.
- Make sure function is recognized by Mantid, you can check the overall fitting by using the following script: 
```# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

#input workspace
iw=Load("iris26176_graphite002_iqt")

# Specify the fitting interval
startX=0.002
endX=0.15

#spectrum number
dec=CreateDetectorTable(iw)
spec_number=0

#initial parameters and attributes
q = dec.row(spec_number)['Q elastic']
amp=1
tau1=0.05
gamma=10

fit_out = Fit(InputWorkspace='iw', WorkspaceIndex=spec_number, \
           StartX = startX, EndX=endX, Output='fit_out', \
           Function="name=TeixeiraWaterIqt,Q="+str(q)+",a=0.98,"+\
                    "Amp="+str(amp)+",Tau1="+str(tau1)+",Gamma="+str(gamma) \
                    +";name=LinearBackground,A0=0,A1=-0.154131",
           Minimizer="Levenberg-Marquardt,Damping=0,Verbose=0,AbsError=0.0001,RelError=0.0001")
       
        
costFuncVal=fit_out.OutputChi2overDoF
print("Current cost function is", costFuncVal)
amp=fit_out.OutputParameters.row(0)['Value'] 
tau1=fit_out.OutputParameters.row(1)['Value']
gamma=fit_out.OutputParameters.row(2)['Value']
print('Amplitude: ',amp,' Tau1: ',tau1,' Gamma: ',gamma)
        
fig, axes = plt.subplots(edgecolor='#ffffff', figsize=[8, 8], num='test', subplot_kw={'projection': 'mantid'})
axes.plot(fit_out.OutputWorkspace, color='#1f77b4', label='fit_out_Workspace: Data', wkspIndex=0)
axes.plot(fit_out.OutputWorkspace, color='#ff7f0e', label='fit_out_Workspace: Calc', wkspIndex=1)
axes.plot(fit_out.OutputWorkspace, color='#2ca02c', label='fit_out_Workspace: Diff', wkspIndex=2)
axes.set_title('Fitting results on iris26176_graphite002_iqt')
axes.set_xlabel('Time ($ns$)')
axes.set_ylabel('Fs(Q,t)')
legend = axes.legend(fontsize=8.0).set_draggable(True).legend

fig.show()
# Use plt.show() if running the script outside of Workbench
#plt.show()
# Scripting Plots in Mantid:
# https://docs.mantidproject.org/tutorials/python_in_mantid/plotting/02_scripting_plots.html
```

*Added release notes*

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by thomashampson** [comment]
pre-commit.ci run

**Comment by SilkeSchomann** [comment]
pre-commit.ci run

**Review by SilkeSchomann — DISMISSED** [review]
Both code and documentation look good. The new fitting function seems to work well 👍

File: Framework/PythonInterface/plugins/functions/TeixeiraWaterIqt.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2007 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (rbauststfc) [inline Framework/PythonInterface/plugins/functions/TeixeiraWaterIqt.py:3]: Just a quick query before I gatekeeper this - should the copyright year on this file and the new test file for the function be 2024?

Comment (adriazalvarez) [reply Framework/PythonInterface/plugins/functions/TeixeiraWaterIqt.py]: Ahh.. you are right ! Thanks for noticing it. It is updated now.

**Review by SilkeSchomann — APPROVED** [review]
I am happy to approve this again.

**Pull request merged**